amethyst_network/src/simulation/resource.rs

```diff
@@ -0,0 +1,192 @@
+use super::{
+    client::Client,
+    message::Message,
+    requirements::{DeliveryRequirement, UrgencyRequirement},
+    transport::socket::Socket,
+};
+use std::{collections::VecDeque, net::SocketAddr};
+
+/// Resource serving as the owner of the underlying socket and the queue of messages to be sent.
+pub struct NetworkSimulationResource<S: Socket> {
+    pub(crate) socket: Option<S>,
```

Comment:
I prefer to have this since we do use that at different places. 

```rust
pub(crate) fn get_socket_mut() -> &mut S { }
```

But that is up to you since we only call it internally.

Reply:
You're probably right. Especially because this will likely have to be made public for people to write their own protocol system implementations